kornia/x/trainer.py

```diff
@@ -95,19 +95,19 @@ def fit_epoch(self, epoch: int) -> None:
         self.model.train()
         losses = AverageMeter()
         for sample_id, sample in enumerate(self.train_dataloader):
-            source, target = sample  # this might change with new pytorch dataset structure
+            sample = {"input": sample[0], "target": sample[1]}  # new dataset api will come like this
             self.optimizer.zero_grad()
-
             # perform the preprocess and augmentations in batch
-            img = self.preprocess(source)
-            img = self.augmentations(img)
+            sample = self.preprocess(sample)
+            sample = self.augmentations(sample)
+            sample = self.on_before_model(sample)
             # make the actual inference
-            output = self.model(img)
-            loss = self.criterion(output, target)
+            output = self.model(sample["input"])
+            loss = self.criterion(output, sample["target"])
```

Comment:
in that case in the `ObjectDetectorTrainer` you might need to overrided this part of the code. Besides, could you point what you return ? because to me seems totally wrong return a loss from the model. Another case would be that intermediate features are returned from the model and then we need a special criterion that selects the needed things and computes the actual loss.